src/frameworks/interfaces.ts

```diff
@@ -79,24 +85,32 @@ export interface Framework {
     target: string,
     hostingEmulatorInfo?: EmulatorInfo
   ) => Promise<RequestHandler>;
-  ɵcodegenPublicDirectory: (
-    dir: string,
-    dest: string,
-    target: string,
-    context: {
-      project: string;
-      site: string;
-    }
-  ) => Promise<void>;
+  ɵcodegenPublicDirectory: (dir: string, options?: CodegenPublicDirectoryOptions) => Promise<void>;
   ɵcodegenFunctionsDirectory?: (
     dir: string,
-    dest: string,
-    target: string
+    options?: CodegenFunctionsDirectoryOptions
   ) => Promise<NodeJSFramework | PythonFramework>;
   getValidBuildTargets?: (purpose: BUILD_TARGET_PURPOSE, dir: string) => Promise<string[]>;
   shouldUseDevModeHandle?: (target: string, dir: string) => Promise<boolean>;
 }
 
+export type CodegenFunctionsDirectoryOptions = {
+  dest?: string;
+  target?: string;
+  configuration?: string;
+  frameworksBackend?: HostingBase["frameworksBackend"];
+};
+
+export type CodegenPublicDirectoryOptions = {
+  dest?: string;
+  target?: string;
+  context?: {
+    project: string;
+    site: string;
+  };
+  frameworksBackend?: HostingBase["frameworksBackend"];
+};
```

Comment:
can't we make `options`(the function param), `dest`, `target` and `context` required as they were before? That way we won't need to check and throw in every framework as those fields were already required. I think it should be covered by the type definition